app/models.py

```diff
@@ -344,13 +363,32 @@ def is_public(self):
     @property
     def is_agency(self):
         """
-        Checks to see if the current user is an agency user
+        Check to see if the current user is an agency user.
 
         AGENCY_USER = 'Saml2In:NYC Employees'
 
         :return: Boolean
         """
-        return self.auth_user_type in user_type_auth.AGENCY_USER_TYPES and self.agency_ein is not None
+        return self.auth_user_type in user_type_auth.AGENCY_USER_TYPES and self.agencies is not None
+
+    @property
+    def default_agency_ein(self):
+        """
+        Return the Users default agency ein.
+        :return: String
+        """
+        agency = AgencyUsers.query.join(Users).filter(AgencyUsers.is_primary_agency == True).one_or_none()
```

Comment:
The idea behind this query is to pull the one (and only one) result from the AgencyUsers table that has `is_primary_agency == True` for the current user. 

It is undefined behavior at this point if a User has multiple `Primary Agencies`